src/subclass/interface.rs

```diff
@@ -0,0 +1,241 @@
+    ///
+    /// This is called right after the type was registered and allows
+    /// interfaces to do additional type-specific initialization, e.g.
+    /// for adding prerequisites.
+    ///
+    /// Optional
+    fn type_init(_type_: &mut InitializingType<Self>) {}
+
+    /// Interface initialization.
+    ///
+    /// This is called after `type_init` and before the first implementor
+    /// of the interface is created. Interfaces can use this to do interface-
+    /// specific initialization, e.g. for installing properties or signals
+    /// on the interface, and for setting default implementations of interface
+    /// functions.
+    ///
+    /// Optional
+    fn interface_init(&mut self) {}
+}
+
+pub trait ObjectInterfaceExt: ObjectInterface {
+    /// Install properties on the interface.
+    ///
+    /// All implementors of the interface must provide these properties.
+    fn install_properties(&mut self, properties: &[Property]) {
+        if properties.is_empty() {
+            return;
+        }
+
+        for property in properties {
+            let pspec = (property.1)(property.0);
+            unsafe {
+                gobject_ffi::g_object_interface_install_property(
+                    self as *mut Self as *mut _,
+                    pspec.to_glib_none().0,
+                );
+            }
+        }
+    }
+
+    /// Add a new signal to the interface.
+    ///
+    /// This can be emitted later by `glib::Object::emit` and external code
+    /// can connect to the signal to get notified about emissions.
+    fn add_signal(&mut self, name: &str, arg_types: &[Type], ret_type: Type) {
+        unsafe {
+            super::types::add_signal(
+                *(self as *mut _ as *mut ffi::GType),
+                name,
+                arg_types,
+                ret_type,
+            );
+        }
+    }
+
+    /// Add a new signal with accumulator to the interface.
+    ///
+    /// This can be emitted later by `glib::Object::emit` and external code
+    /// can connect to the signal to get notified about emissions.
+    ///
+    /// The accumulator function is used for accumulating the return values of
+    /// multiple signal handlers. The new value is passed as second argument and
+    /// should be combined with the old value in the first argument. If no further
+    /// signal handlers should be called, `false` should be returned.
+    fn add_signal_with_accumulator<F>(
+        &mut self,
+        name: &str,
+        arg_types: &[Type],
+        ret_type: Type,
+        accumulator: F,
+    ) where
+        F: Fn(&mut Value, &Value) -> bool + Send + Sync + 'static,
+    {
+        unsafe {
+            super::types::add_signal_with_accumulator(
+                *(self as *mut _ as *mut ffi::GType),
+                name,
+                arg_types,
+                ret_type,
+                accumulator,
+            );
+        }
+    }
+
+    /// Add a new action signal with accumulator to the interface.
```

Comment:
With accumulator?

Reply:
What do you mean? :)

Also this is the same documentation as we already had for subclasses before

Reply:
I missed that `handler` actually `GSignalAccumulator accumulator`, sorry